### PR TITLE
refactor: refactored alerting

### PR DESCRIPTION
**Pull request by phbelitz**
Recfatored the alerting feature

### CODE REVIEW

File: connaisseur/flask_server.py [hunk 1]
```diff
@@ -60,8 +60,7 @@ def mutate():
         else:
             err_log = str(traceback.format_exc())
             msg = "unknown error. please check the logs."
-        if call_alerting_on_request(admission_request, admitted=False):
-            send_alerts(admission_request, admitted=False, reason=msg)
+        AlertingConfiguration().send_alerts(admission_request, False, msg)
```
Comment (Starkteetje) [inline connaisseur/flask_server.py:63]: This reads weirdly. A configuration should not send alerts. Either the Config should be named differently or it has too many capabilities

File: connaisseur/tests/test_alert.py [hunk 2]
```diff
@@ -2,13 +2,8 @@
...
 
-from connaisseur.alert import (
-    Alert,
-    send_alerts,
-    call_alerting_on_request,
-    get_alert_config_validation_schema,
-    load_config,
-)
+import conftest as fix
+import connaisseur.alert as al
```
Comment (annekebr) [inline connaisseur/tests/test_alert.py:6]: Is there a reason for  not importing the functions themselves? The prefix `al` doesn't really tell anything so if we don't want to import the functions/classes themselves I'd go for plain `alert` instead for better readability.

File: connaisseur/alert.py [hunk 3]
```diff
@@ -5,23 +5,98 @@
...
+            self.config = {}
+        except InvalidConfigurationFormatError as err:
+            raise ConfigurationError(err.message) from err
+        except Exception as err:
+            raise ConfigurationError(
+                "An error occurred while loading the AlertingConfiguration file:"
+                f"{str(err)}"
+            ) from err
+
+    def __alerting_required(
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply connaisseur/alert.py]: split this up into `AlertingConfiguration.alerting_required` and `__is_not_hook_image`

File: connaisseur/tests/conftest.py [hunk 4]
```diff
@@ -367,20 +393,25 @@ def adm_req_samples(m_ad_schema_path):
 
 
 @pytest.fixture()
-def mock_safe_path_func_load_config(mocker):
+def m_safe_path_func(monkeypatch):
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply connaisseur/tests/conftest.py:396]: i named all the fixtures `m_*` and the fuctions inside the fixture (if present) `mock_*`. this is just to keep things consistent with the rest of the codebase.

File: connaisseur/tests/data/alerting/alertconfig.json [hunk 5]
```diff
@@ -40,7 +40,7 @@
                 "custom_headers": [
                     "Content-Language: de-DE"
                 ],
-                "fail_if_alert_sending_fails": true,
+                "fail_if_alert_sending_fails": false,
```
[the remark this answers is not in the document]
Comment (phbelitz) [reply connaisseur/tests/data/alerting/alertconfig.json:43]: if this is `true` the second test of `test_mutate` in `test_flask_server.py` will fail, as an exception is raised during validation, and no response_object for inspection will be returned. this isn't the goal of the test to check whether the exception gets raised. this is handled by another test.

**Review by annekebr — APPROVED** [review]

**Review by annekebr — APPROVED** [review]

**Pull request merged**